tests/unit_tests/core/fake_device_module.py

```diff
@@ -32,7 +32,7 @@ def _mock_with_name(name: str) -> MagicMock:
     return mock
 
 
-def wrong_return_type() -> int:
+def wrong_return_type(*args, **kwargs) -> int:
```

Comment:
I must admit I can't fully remember why this was needed as it was some weeks ago. I do remember though that the tests would not succeed (when they should do) without this change. It was probably to do with the examination of the funciton  signature by the start_as_current_span decorator

Reply:
I think adding unused, unnecessary parameters to satisfy some library somewhere is not good practice (or in more succinct terms: https://github.com/DiamondLightSource/blueapi/pull/669#discussion_r1804649606), we should try to work out what is complaining and see if we can satisfy it another way. Try removing it and seeing why the tests fail, I'm happy to help out if it's a weird error message.

Reply:
...So the reason this happens is because of the context propagation from the main process to the subprocess, which you have facilitated via changes to `blueapi.service.runner.WorkerDispatcher`.

These changes take the observability context and pass it to the function in the subprocess, along with any arguments that the user supplies. It _assumes_ that the function is decorated with `@use_propagated_context`, but is not very explicit about this. So if you remove `*args, **kwargs` from these functions the tests fail because you are assuming that those functions were decorated with `@use_propagated_context` and thus take a context as an argument.

There are a few possible solutions:
1. Don't require `@use_propagated_context` to have been added to the subprocess function, make it optional and add logic to `WorkerDispatcher` to check whether it's been done and chose whether or not to pass the context.
2. Continue to require `@use_propagated_context` but make sure mypy (or least pyright when #619 is done) complains in cases like this. Then Decorate these functions with `@use_propagated_context` and remove the `*args, **kwargs`. N.b. I'm not actually sure if `ParamSpec` is powerful enough to do this. @DiamondJoseph?
3. Find another way of propagating the context rather than through function arguments. For instance I don't think the functionality in `@use_propagated_context` _needs_ to be in a decorator because it is just setting a global. So we could instead put that logic in `WorkerDispatcher`, but deferred to the subprocess via `self._subprocess.apply`.

My preference is 3. if no one else can see anything wrong with it, because it removes the hidden messing around with function arguments, which could confuse the unsuspecting.

Reply:
Implemented option 3, passing the context info into _rpc and modifiying it to use this to attach this (if not None) as the current context in the subprocess side.

Reply:
Cool, if you do that you should be able to remove `@use_propagated_context` from the `interface.py` functions and the `*args, **kwargs` from these functions.

Reply:
Yes done that too